### ISSUE TITLE
Maintenance clean_old_jobs process race condition

### ISSUE THREAD

**Issue by ahammond** [saltstack/salt#33869]
### Description of Issue/Question
Looks like a race condition around cleaning up old jobs.

```
[DEBUG   ] LazyLoaded local_cache.clean_old_jobs
[ERROR   ] An un-handled exception from the multiprocessing process 'Maintenance-4' was caught:
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/process.py", line 613, in _run
    return self._original_run()
  File "/usr/lib/python2.6/site-packages/salt/master.py", line 236, in run
    salt.daemons.masterapi.clean_old_jobs(self.opts)
  File "/usr/lib/python2.6/site-packages/salt/daemons/masterapi.py", line 187, in clean_old_jobs
    mminion.returners[fstr]()
  File "/usr/lib/python2.6/site-packages/salt/returners/local_cache.py", line 413, in clean_old_jobs
    shutil.rmtree(t_path)
  File "/usr/lib64/python2.6/shutil.py", line 204, in rmtree
    onerror(os.listdir, path, sys.exc_info())
  File "/usr/lib64/python2.6/shutil.py", line 202, in rmtree
    names = os.listdir(path)
OSError: [Errno 2] No such file or directory: '/var/cache/salt/master/jobs/66'
Process Maintenance-4:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 232, in _bootstrap
    self.run()
  File "/usr/lib/python2.6/site-packages/salt/utils/process.py", line 613, in _run
    return self._original_run()
  File "/usr/lib/python2.6/site-packages/salt/master.py", line 236, in run
    salt.daemons.masterapi.clean_old_jobs(self.opts)
  File "/usr/lib/python2.6/site-packages/salt/daemons/masterapi.py", line 187, in clean_old_jobs
    mminion.returners[fstr]()
  File "/usr/lib/python2.6/site-packages/salt/returners/local_cache.py", line 413, in clean_old_jobs
    shutil.rmtree(t_path)
  File "/usr/lib64/python2.6/shutil.py", line 204, in rmtree
    onerror(os.listdir, path, sys.exc_info())
  File "/usr/lib64/python2.6/shutil.py", line 202, in rmtree
    names = os.listdir(path)
OSError: [Errno 2] No such file or directory: '/var/cache/salt/master/jobs/66'
[INFO    ] Process <class 'salt.master.Maintenance'> (27517) died with exit status None, restarting...
```

### Setup
This is a vbox salt master used for development. Upgraded from 2015.8.8.2 where it was running pretty smoothly.

### Steps to Reproduce Issue
Unknown.

### Versions Report
```
[root@salt ~]# salt-call --local test.versions
local:
    Salt Version:
               Salt: 2016.3.0

    Dependency Versions:
               cffi: 1.6.0
           cherrypy: Not Installed
           dateutil: Not Installed
              gitdb: 0.5.4
          gitpython: 0.3.2 RC1
              ioflo: Not Installed
             Jinja2: 2.7.3
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: 0.20.2
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: Not Installed
          pycparser: 2.14
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.5.0
               RAET: Not Installed
              smmap: 0.8.1
            timelib: 0.2.4
            Tornado: 4.2.1
                ZMQ: 4.0.5

    System Versions:
               dist: centos 6.6 Final
            machine: x86_64
            release: 2.6.32-504.30.3.el6.x86_64
             system: Linux
            version: CentOS 6.6 Final
```

**Comment by L4rS6** [comment]
See here: https://github.com/saltstack/salt/pull/33555

**Comment by Ch3LL** [comment]
yep @L4rS6 is definitely right this is fixed by that PR he linked. Also this is a duplicate of #33544. This will be fixed in 2016.3.1 which I believe we are shooting to release next week. I will go ahead and close this since its already been confirmed as fixed. Thanks
